api-private.swagger.yaml

```diff
@@ -1,376 +1,519 @@
+        schema:
           type: string
-          description: The UUID of the node to export.
-          pattern: '^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$'
-          required: true
-        - name: body
-          in: body
-          schema:
-            $ref: '#/definitions/pdf-export-job-submission-schema'
-          required: true
+          format: uuid
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/pdf-export-job-submission-schema'
+        required: true
       responses:
-        '200':
+        200:
           description: An existing PDF export job was found and returned to the client.
-          schema:
-            $ref: '#/definitions/pdf-export-job-response-schema'
-        '201':
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/pdf-export-job-response-schema'
+        201:
           description: A new PDF export job was created.
-          schema:
-            $ref: '#/definitions/pdf-export-job-response-schema'
-        '400':
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/pdf-export-job-response-schema'
+        400:
           description: A malformed UUID was supplied.
-definitions:
-  session-schema:
-    properties:
-      user_id:
-        description: The Invotra UUID of the user
-        type: string
-        pattern: '^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$'
-        example: 2df46a6e-6f73-4b09-b340-7cea68b6e5f6
-      user_internal_id:
-        description: The Drupal user ID
-        type: integer
-        example: 12345
-      expiry:
-        description: The number of seconds after which the session should expire
-        type: integer
-        example: 1209600
-    required:
-      - user_id
+          content: {}
+      x-codegen-request-body-name: body
+  /preview_mode/{user_id}:
+    post:
+      tags:
+        - preview mode
+      summary: "Comunicate that preview mode has been enabled for the specified user's session."
+      parameters:
+        - name: user_id
+          in: path
+          description: 'The Invotra UUID of the user.'
+          required: true
+          schema:
+            type: string
+            format: uuid
+      x-codegen-request-body-name: body
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/preview-mode-submission-schema' 
+      responses:
+        201:
+          description: "Acknowledged that preview mode for the user's session has been enabled."
+          content: {}
+        400:
+          description: 'A malformed UUID was supplied.'
+          content: {}
+        404:
+          description: 'The user was not found.'
+          content: {}
+    delete:
```

Comment:
should this be a POST to include the session?

Reply:
made the DELETE a PUT with session body